File: server/ruby/send-template-to-tenant.rb
require 'dotenv'
require 'trycourier'

env_path = File.join(File.dirname(__FILE__), '..', '.env')
Dotenv.load(env_path)

api_key = ENV['COURIER_API_KEY']
tenant_id = ENV['COURIER_SEND_TEMPLATE_TO_TENANT_TENANT_ID']
template_id = ENV['COURIER_SEND_TEMPLATE_TO_TENANT_ID_TEMPLATE_ID']

client = Trycourier::Client.new(api_key: api_key)

response = client.send_.message(
  message: {
    to: {
      tenant_id: tenant_id
    },
    template: template_id
  }
)

puts response

File: server/php/send-template-to-tenant.php
<?php
require_once __DIR__ . '/vendor/autoload.php';
use Courier\Client;
use Dotenv\Dotenv;

$dotenv = Dotenv::createImmutable(__DIR__ . '/..');
$dotenv->load();

$api_key = $_ENV['COURIER_API_KEY'] ?? '';
$tenant_id = $_ENV['COURIER_SEND_TEMPLATE_TO_TENANT_TENANT_ID'] ?? '';
$template_id = $_ENV['COURIER_SEND_TEMPLATE_TO_LIST_TEMPLATE_ID'] ?? '';

$client = new Client(apiKey: $api_key);

$response = $client->send->message([
    'message' => [
        'to' => [
            'tenant_id' => $tenant_id
        ],
        'template' => $template_id
    ]
]);

echo $response . "\n";
